File: KalkJava/Kalk/FunctionResult.java
package Kalk;

import java.util.*;

public class FunctionResult extends Result {
	private ArrayList<Pair<Double, Double>> points = new ArrayList<Pair<Double, Double>>();

	public FunctionResult(ArrayList<Pair<Double, Double>> v) {
		super();
		this.points = new ArrayList<Pair<Double, Double>>(v);
	}
	public FunctionResult(FunctionResult f) {
		super();
		this.points = new ArrayList<Pair<Double, Double>>(f.points);
	}

	public final ArrayList<Pair<Double, Double>> getPoints() {
		return points;
	}

	@Override
	public final void stampa() {
		System.out.print("X\tY");
		System.out.print("\n");

		for(int i = 0; i < points.size(); i++) {
			System.out.print(points.get(i).first);
			System.out.print("\t");
			System.out.print(points.get(i).second);
			System.out.print("\t");
			System.out.print("\n");
		}
	}

	public final FunctionResult copyFrom(FunctionResult f) {
		if(this != f) {
			points = new ArrayList<Pair<Double, Double>>();
			for(Pair<Double, Double> i : f.points) {
				points.add(new Pair<Double, Double>(i.first,i.second));
			}
		}
		return this;
	}

	public double toDouble() {
		double s = 0;
		for(Pair<Double, Double> i : points) {
			s += i.second;
		}
		s /= points.size();
		return s;
	}
}

File: Kalk/model/functionresult.cpp
#include "functionresult.h"

FunctionResult::FunctionResult(const std::vector<std::pair<double, double>> v) : Result(), points(v.begin(), v.end()) {}

FunctionResult::FunctionResult(const FunctionResult& f) : Result(), points(f.points) {}

FunctionResult& FunctionResult::operator =(const FunctionResult& f)
{
    if(this != &f) {
        points = std::vector<std::pair<double, double>>(f.points.begin(), f.points.end());
    }
    return *this;
}

FunctionResult::operator double() const
{
    double s = 0;
    for (auto i = points.begin(); i != points.end(); ++i)
        s += i->second;
    s /= points.size();
    return s;
}

std::vector<std::pair<double, double>> FunctionResult::getPoints() const
{
    return points;
}

void FunctionResult::stampa() const
{
    std::cout << "X\tY" << std::endl;
    for(unsigned int i = 0; i < points.size(); i++){
        std::cout << points[i].first << "\t";
        std::cout << points[i].second << "\t";
        std::cout << std::endl;
    }
}

